x-pack/plugins/ml/public/application/components/scatterplot_matrix/scatterplot_matrix.tsx

```diff
@@ -172,7 +181,77 @@ export const ScatterplotMatrix: FC<ScatterplotMatrixProps> = ({
     setDynamicSize(!dynamicSize);
   };
 
-  const { euiTheme } = useCurrentEuiTheme();
+  const getCustomVisualizationLink = useCallback(() => {
+    const { columns } = splom!;
+    const outlierScoreField =
+      resultsField !== undefined ? `${resultsField}.${OUTLIER_SCORE_FIELD}` : undefined;
+    const vegaSpec = getScatterplotMatrixVegaLiteSpec(
+      true,
+      [],
+      [],
+      columns,
+      euiTheme,
+      resultsField,
+      color,
+      legendType,
+      dynamicSize
+    );
+
+    vegaSpec.$schema = 'https://vega.github.io/schema/vega-lite/v5.json';
+    vegaSpec.title = `Scatterplot matrix for ${index}`;
+
+    const fieldsToFetch = [
+      ...columns,
+      // Add outlier_score field in fetch if it's available so custom visualization can use it
+      ...(outlierScoreField ? [outlierScoreField] : []),
+      // Add field to color code by in fetch so custom visualization can use it -  usually for classfication jobs
+      ...(color ? [color] : []),
+    ];
+
+    vegaSpec.data = {
+      url: {
+        '%context%': true,
+        ...(indexPattern?.timeFieldName
+          ? { ['%timefield%']: `${indexPattern?.timeFieldName}` }
+          : {}),
+        index,
+        body: {
+          fields: fieldsToFetch,
+          size: 1000,
```

Comment:
Can we apply the value from the UI dropdown to select the amount of docs to fetch?

Reply:
Good catch - updated in e14f32ff3f779436910ec8787c521251916a7972